pytest import fails if kubeconfig has invalid OIDC token

Running: 
`make test tests=src/toil/test/sort/sortTest.py::SortTest::testSort`
or 
`python -m pytest src/toil/test/sort/sortTest.py::SortTest::testSort -vv`

fails during import of `src/toil/test/conftest.py` because `src/toil/test/__init__.py` calls `_is_kubernetes_installed_and_configured(),` which calls `kubernetes.config.load_kube_config()`.

Error:
`E   oauthlib.oauth2.rfc6749.errors.CustomOAuth2Error: ("invalid_token"
E   error_description="token not found"
E   )`

I likely have a staled/revoked OIDC token in `~/.kube/config` (kubectl/kubeconfig present, but creds not valid).
Expected behavior is that this is treated as “kubernetes not usable” and Kubernetes tests are skipped, instead of failing pytest import.

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/TOIL-1829)
┆Issue Number: TOIL-1829

--- a/src/toil/test/__init__.py
+++ b/src/toil/test/__init__.py
@@ -529,6 +529,7 @@ def needs_kubernetes_installed(test_item: MT) -> MT:
 def _is_kubernetes_installed_and_configured() -> bool:
     try:
         import kubernetes
+        import oauthlib.oauth2.rfc6749.errors  # type: ignore[import-untyped]
 
         try:
             kubernetes.config.load_kube_config()  # type: ignore[attr-defined]
@@ -537,7 +538,13 @@ def _is_kubernetes_installed_and_configured() -> bool:
                 kubernetes.config.load_incluster_config()  # type: ignore[attr-defined]
             except kubernetes.config.ConfigException:  # type: ignore[attr-defined]
                 return False
-    except ImportError:
+        except oauthlib.oauth2.rfc6749.errors.CustomOAuth2Error:
+            logger.warning("Kubernetes credentials are not accepted by the cluster.")
+            return False
+        except Exception:
+            logger.exception("Unexpected error while checking for Kubernetes configuration.")
+            return False
+    except ImportError: # to catch any exception 
         return False
     return True
 

--- a/src/toil/test/__init__.py
+++ b/src/toil/test/__init__.py
@@ -541,10 +541,10 @@ def _is_kubernetes_installed_and_configured() -> bool:
         except oauthlib.oauth2.rfc6749.errors.CustomOAuth2Error:
             logger.warning("Kubernetes credentials are not accepted by the cluster.")
             return False
-        except Exception:
+        except Exception: # to catch any exception
             logger.exception("Unexpected error while checking for Kubernetes configuration.")
             return False
-    except ImportError: # to catch any exception 
+    except ImportError:
         return False
     return True
 

